lib/miam/client.rb

```diff
@@ -266,6 +266,21 @@ def walk_assume_role_policy(role_name, expected_assume_role_policy, actual_assum
     expected_assume_role_policy.sort_array!
     actual_assume_role_policy.sort_array!
 
+    # With only one entity granted
+    # On IAM
+    #   (1) Statement => [ { Principal => AWS => arn } ]
+    # Should be able to specify like:
+    #   (2) Statement => [ { Principal => AWS => [arn] } ]
+    # Actually (1) is reflected when config (2) is applied
+    if expected_assume_role_policy.key?('Statement')
+      expected_assume_role_policy['Statement'].each do |stmt|
+        next unless stmt.key?('Principal')
+        stmt['Principal'].each do |k, v|
+          stmt['Principal'][k] = v[0] if v.is_a?(Array) && v.length < 2
+        end
+      end
+    end
+
```

Comment:
almost ok 👍 
but it is not good to update hash in`each` method.
so I think that the better to use `keys` method and temporary variable.

```ruby
principal = stmt['Principal']
principal.keys.each do |k|
  entities = Array(principal[k])
  principal[k] = entities.first if entities.length < 2
end
```